### PR TITLE
Temporary move `rememberComposableLambda` from commonMain to jvmAndNativeMain

### PR DESCRIPTION
**Pull request by eymar**
**Description:**
Currently, compose compiler plugin doesn't handle properly this function when targeting web.

We will move the function back to commonMain, after all compiler plugins support the new function in web targets.

Fixes CMP-1547 for desktop and k/native. Web is still affected by the issue.

## Testing
1. Open :compose:mpp:demo on desktop
2. Go to Components -> Dialogs -> Dialog with text field
3. Close the dialog and Go back (press the back button)
No unexpected dialog is shown.

To reproduce the issue, comment `rememberComposableLambda`.
Repeat the steps above.
An unexpected dialog will be shown after back button is pressed.

**Note**: when we move the function back, it means that CM 1.7 projects won't be able to use kotlin 2.0 with web targets. Only kotlin 2.0.10 if the compiler fix is accepted.

### CODE REVIEW

**Review by MatkovIvan — APPROVED** [review]

**Review by igordmn — APPROVED** [review]

File: compose/runtime/runtime/src/jvmAndNativeMain/kotlin/androidx/compose/runtime/internal/ComposableLambda.jvmAndNative.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmName("ComposableLambdaKt")
+@file:JvmMultifileClass
```
Comment (eymar) [inline compose/runtime/runtime/src/jvmAndNativeMain/kotlin/androidx/compose/runtime/internal/ComposableLambda.jvmAndNative.kt:18]: this ensures backward compatibility when we move the funciton back.

The generated bytecode contains a new class anyway, it's named - `ComposableLambda_jvmAndNativeKt_ComposableLambdaKt`. It's not used directly by anything except the original ComposableLambdaKt.

In the generated bytecode, the original ComposableLambdaKt contains `rememberComposableLambda` too, but it delegates to `ComposableLambdaKt__ComposableLambda_jvmAndNativeKt`:

```java
@ComposeCompilerApi
   @Composable
   @NotNull
   public static final ComposableLambda rememberComposableLambda(int key, boolean tracked, @NotNull Object block, @Nullable Composer $composer, int $changed) {
      return ComposableLambdaKt__ComposableLambda_jvmAndNativeKt.rememberComposableLambda(key, tracked, block, $composer, $changed);
   }
   ```

File: compose/runtime/runtime/build.gradle [hunk 2]
```diff
@@ -175,6 +175,11 @@ if(AndroidXComposePlugin.isMultiplatformEnabled(project)) {
                 }
             }
 
+            def jvmAndNativeMain = sourceSets.create("jvmAndNativeMain")
+            jvmAndNativeMain.dependsOn(commonMain)
```
Comment (igordmn) [inline compose/runtime/runtime/build.gradle:179]: maybe `commonNonWebMain` as we named it in another PR?

Comment (eymar) [reply compose/runtime/runtime/build.gradle:179]: do you mean `commonNonWebMain`, since it's not a test source set?

Comment (MatkovIvan) [reply compose/runtime/runtime/build.gradle:179]: Maybe nonWebMain? Like nonJvmMain that we already have

**Comment by eymar** [comment]
The change needed in compose compiler for web was already merged to kotlin 2.0.10 and master branches
Therefore, this PR became obsolete.

We'll update the compose compiler plugin for 1.9.2x too and then web should work the same as desktop and ios.

**Pull request closed without merging**